src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceChannelFactory.cs

```diff
@@ -471,28 +478,25 @@ protected override void OnClose(TimeSpan timeout)
                 _innerChannelFactory.Close(timeoutHelper.RemainingTime());
             }
 
+
             protected override IAsyncResult OnBeginClose(TimeSpan timeout, AsyncCallback callback, object state)
             {
-                return new ChainedAsyncResult(timeout, callback, state, base.OnBeginClose, base.OnEndClose,
-                    _innerChannelFactory.BeginClose, _innerChannelFactory.EndClose);
+                throw ExceptionHelper.PlatformNotSupported();   // Issue #31 in progress
+                //return new ChainedAsyncResult(timeout, callback, state, base.OnBeginClose, base.OnEndClose,
+                //    _innerChannelFactory.BeginClose, _innerChannelFactory.EndClose);
             }
 
             protected override void OnEndClose(IAsyncResult result)
             {
-                ChainedAsyncResult.End(result);
+                throw ExceptionHelper.PlatformNotSupported();   // Issue #31 in progress
+                //ChainedAsyncResult.End(result);
             }
 
             protected internal override Task OnCloseAsync(TimeSpan timeout)
             {
                 return OnCloseAsyncInternal(timeout);
             }
 
-            protected internal override Task OnOpenAsync(TimeSpan timeout)
-            {
-                this.OnOpen(timeout);
-                return TaskHelpers.CompletedTask();
-            }
-
```

Comment:
Great question -- I did remove it deliberately.  This base ServiceChannelFactory class does not override OnBeginOpen. It appears the TypedServiceChannelFactory does the overrides and does not call base, and therefore this became dead code.  By not overriding here, if it is ever called it will fall all the way through to CommunicationObject where an Assert will fire to alert us we need this override here too.  That's part of the pattern to move to the new async open/close.  Any type that *needs* to override but does not triggers the Assert, and then we go fix it.  I relied on that a lot in this PR.

---
In reply to: [78156592](https://github.com/dotnet/wcf/pull/1515#discussion_r78156592) [](ancestors = 78156592)